sdk/core/System.ClientModel/gen/ModelReaderWriterContextGenerator.cs

```diff
@@ -166,6 +166,12 @@ private void ReportDiagnosticAndEmitSource(
                     return null;
 
                 var type = data.SymbolToTypeRefCache.Get(typeSymbol, data.SymbolToKindCache);
+                if (type.ObsoleteLevel == ObsoleteLevel.Error)
+                {
+                    // if its marked as error obsolete we can't create a builder for it
```

Comment:
Can we expand a bit on the context of why this is true just for future reference?